File: ios/stereoCamera/stereoCamera/Comm/Commands/LatencyTest.swift
//
//  LatencyTest.swift
//  stereoCamera
//
//  Created by hallmarklabs on 8/17/18.
//  Copyright © 2018 cody. All rights reserved.
//

import Foundation

class LatencyTest : Command
{
    var elapsed:Int = 0
    
    override init()
    {
        super.init()
        cmdtype = CommandTypes.LATENCY_CHECK
        expectsResponse = true
    }
    
    override func send(comm: Comm) -> Bool
    {
        let result = super.send(comm: comm)
        if (!result) { return false }
        
        var bytes:[UInt8] = []
        bytes += Bytes.toByteArray(elapsed)
        
        let sz = comm.write(buf: bytes)
        return (sz <= 0) ? false : true
    }
    
    override func receive(comm: Comm) -> Bool
    {
        let result = super.receive(comm: comm)
        if (!result) { return false }
        
        let (buf2, sz2) = comm.read(sz:8)
        if (sz2 <= 0) { return false }
        
        elapsed = Bytes.fromByteArray(buf2)
        
        return true
    }
}

File: android/app/src/main/java/com/munger/stereocamera/ip/command/commands/LatencyTest.java
package com.munger.stereocamera.ip.command.commands;

import com.munger.stereocamera.ip.command.Comm;
import com.munger.stereocamera.ip.command.Command;

import java.io.IOException;

public class LatencyTest extends Command
{
	public long elapsed = 0;

	public LatencyTest()
	{
		super();
		cmdtype = Type.LATENCY_CHECK;
		expectsResponse = true;
	}

	@Override
	public boolean send(Comm comm)
	{
		boolean result = super.send(comm);
		if (!result)
			{ return false; }

		try
		{
			comm.putLong(elapsed);
		}
		catch(IOException e){
			return false;
		}

		return true;
	}

	@Override
	public boolean receive(Comm comm)
	{
		boolean result = super.receive(comm);
		if (!result)
			{ return false; }

		try
		{
			elapsed = comm.getLong();
		}
		catch(IOException e){
			return false;
		}

		return true;
	}
}
